plugins/catalog/src/components/CatalogTable/CatalogTable.tsx

```diff
@@ -153,23 +153,25 @@ export const CatalogTable = ({
   const actions: TableProps<EntityRow>['actions'] = [
     ({ entity }) => {
       const location = findLocationForEntityMeta(entity.metadata);
+      const url = findViewUrl(entity, location);
       return {
         icon: () => <OpenInNew fontSize="small" />,
         tooltip: 'View',
         onClick: () => {
-          if (!location) return;
-          window.open(location.target, '_blank');
+          if (!url) return;
+          window.open(url, '_blank');
         },
       };
     },
     ({ entity }) => {
       const location = findLocationForEntityMeta(entity.metadata);
+      const url = findEditUrl(entity, location);
       return {
         icon: () => <Edit fontSize="small" />,
```

Comment:
We mentioned disabling these when there was no url to go to, right? I am not sure how to best make it visually disabled, but atleast the tooltip should tell you why you can't click it. Same above

Reply:
I think that as currently implemented, there will always be a URL, since if all else fails it will just fall back to `location.target`. `location` is hypothetically nullable, but I can't see how you'd get there without manually editing the database ... 

Cleanly fixing that I think would mean doing what @Fox32 suggests and moving the logic to the backend/processor step.

Reply:
But location targets are not always even URLs at all. They could even be the empty string. We can make no assumptions about them whatsoever actually. So we could even go as far as running it through the URL() constructor and return undefined if it throws, or if the protocol was neither http nor https.